integration_tests/src/main/python/date_time_test.py

```diff
@@ -459,10 +459,12 @@ def test_to_timestamp(parser_policy):
             .select(f.col("a"), f.to_timestamp(f.col("a"), "yyyy-MM-dd HH:mm:ss")),
         { "spark.sql.legacy.timeParserPolicy": parser_policy})
 
+# mm: minute; MM: month
 @pytest.mark.skipif(not is_supported_time_zone(), reason="not all time zones are supported now, refer to https://github.com/NVIDIA/spark-rapids/issues/6839, please update after all time zones are supported")
 @pytest.mark.parametrize("format", ['yyyyMMdd', 'yyyymmdd'], ids=idfn)
 def test_formats_for_legacy_mode(format):
-    gen = StringGen("[0-9]{3}[1-9](0[1-9]|1[0-2])(0[1-9]|[1-2][0-9])")
+    # this regexp excludes zero year, python does not like zero year
+    gen = StringGen("([0-9]{3}[1-9])([0-5][0-9])([0-3][0-9])")
```

Comment:
This makes me wonder what will happen if an illegal input is given. We may need to test this case.

Reply:
Illegal input produces NULL.
Added a test case which generates random 8 digits strings(exclude year 0000)